ruby.c

```diff
@@ -1819,8 +1819,10 @@ ruby_opt_init(ruby_cmdline_options_t *opt)
 
     if (rb_namespace_available())
         rb_initialize_main_namespace();
+    rb_namespace_init_done();
```

Comment:
For anyone curious, this function only sets a flag:

```c
void
rb_namespace_init_done(void)
{
    ruby_namespace_init_done = true;
}
```

So moving it before JIT initialization is probably fine.

Reply:
What it does is that any class defined before this point use a larger object slot, so that it can have multiple `classext_t`.

And yes, I think moving it here is fine.